packages/migration/src/migrations/user-mgnt/20231106130956-email-mobile-unique-index.ts

```diff
@@ -15,6 +15,40 @@ export const up = async (db: Db, client: MongoClient) => {
   const session = client.startSession()
   try {
     await session.withTransaction(async () => {
+      console.log('Checking if mobile or email field already have indices')
+      const indexes = Object.keys(
+        (await db.collection('users').stats()).indexSizes
+      )
+      const mobileHasIndex = indexes.some(
+        (index) => index.split('_')[0] === ('mobile' as const)
+      )
+      const emailForNotificationHasIndex = indexes.some(
+        (index) => index.split('_')[0] === ('emailForNotification' as const)
+      )
+
+      if (mobileHasIndex) {
+        console.log('Removing index of mobile')
+        await db.collection('users').dropIndex({ mobile: 1 } as any)
+      }
+
+      if (emailForNotificationHasIndex) {
+        console.log('Removing index of email')
+        await db
+          .collection('users')
+          .dropIndex({ emailForNotification: 1 } as any)
+      }
+      // Ensure that the "mobile" field in the "users" collection does not contain null or duplicate values
+      // Before making the field sparse, which allows it to have null values but ensures uniqueness
+      const updateResult = await db.collection('users').updateMany(
+        { mobile: { $in: [null, ''] } },
+        {
+          $unset: {
+            mobile: ''
+          }
+        }
+      )
+      console.log('Number of documents updated:', updateResult.modifiedCount)
```

Comment:
What about the dropped emailForNotification index?  Only mobile is recreated here

Reply:
Actually, we don't need to drop the index of `emailForNotification` and it should not contain any null values in any version. So I think it's okay to remove the `dropIndex` of `emailForNotification`. Thus, wouldn't need the count of that.